compiler/src/dotty/tools/dotc/typer/Dynamic.scala

```diff
@@ -215,10 +216,7 @@ trait Dynamic {
       errorTree(tree, em"Structural access not allowed on method $name because it $reason")
 
     fun.tpe.widen match {
-      case tpe: ValueType =>
-        structuralCall(nme.selectDynamic, Nil).cast(tpe)
-
-      case tpe: MethodType =>
+      case tpe @ (_: MethodType | _: ErasedValueType) =>
```

Comment:
@odersky could you confirm this change should work and not break something else?